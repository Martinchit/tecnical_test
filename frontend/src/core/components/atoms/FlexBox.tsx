import styled from 'styled-components';

export interface FlexBoxProps {
  flexDirection?: string;
  flexWrap?: string;
  justifyContent?: string;
  alignItems?: string;
  width?: string;
  height?: string;
  margin?: string;
  padding?: string;
}

export const FlexBox = styled.div<FlexBoxProps>`
  display: flex;
  flex-direction: ${({ flexDirection }) => flexDirection ?? 'row'};
  flex-wrap: ${({ flexWrap }) => flexWrap ?? 'wrap'};
  justify-content: ${({ justifyContent }) => justifyContent ?? 'center'};
  align-items: ${({ alignItems }) => alignItems ?? 'center'};
  width: ${({ width }) => width ?? '100%'};
  height: ${({ height }) => height ?? '100%'};
  margin: ${({ margin }) => margin ?? 0};
  padding: ${({ padding }) => padding ?? 0};
`;
