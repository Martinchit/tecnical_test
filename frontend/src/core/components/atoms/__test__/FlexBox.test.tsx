import React from 'react';
import { shallow } from 'enzyme';

import { FlexBox } from '../FlexBox';

describe('FlexBox', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<FlexBox />);
    expect(wrapper).toMatchSnapshot();
  });
});
