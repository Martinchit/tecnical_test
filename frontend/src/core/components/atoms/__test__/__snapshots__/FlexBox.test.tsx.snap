// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexBox renders correctly 1`] = `
<div
  className="sc-AxjAm cUxzLf"
/>
`;
